### ISSUE TITLE
Automatically open sidebar for current folder if the screen size allows it

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/server#7138]
This is something we talked about at the conference: If the screen width is above a certain value, we could automatically open the sidebar for the folder you are currently in.

That would help make the things we offer in the sidebar like comments and sharing more obvious and directly accessible.

The only question is what we do in the root folder. For the first iteration we could simply not show the sidebar. In the future we could show the activity tab and simply show all activity like in the Activity app. cc @nickvergessen 

What do you think @nextcloud/designers?

**Comment by skjnldsv** [comment]
I completely agree! We have far too much empty space on big screens :)

**Comment by jancborchardt** [comment]
How could we do it? I guess we need JS for it @nextcloud/javascript

**Comment by nickvergessen** [comment]
>  In the future we could show the activity tab and simply show all activity like in the Activity app.

Sorry, but please **don't** turn the files app into a portal/dashboard/whatever...

**Comment by jancborchardt** [comment]
- in the sidebar of a file, the activity tab shows activity on the file
- in the sidebar of a folder, ideally activity of the whole folder is shown but is not atm due to technical reasons
- in the root folder, the sidebar could show all activity

It's not about turning the Files app into anything, it's providing info where it's expected or useful.

As said above, as another solution, in the root folder we could also simply not show the sidebar.

**Comment by Espina2** [comment]
In the frontend you can do that with a flag. So you verify with javascript what is the window width, if is bigger than X set the flag to true. 

And you will have something like that, if the flag is true onclick do this. That is something easy todo. 

I am not really sure about showing the bar, it can look weird. Maybe we should do it first and take the decision after. @jancborchardt

**Comment by blizzz** [comment]
Directory activities would recursively get the activities of their children? Sounds expensive, if not pre-generated.

**Comment by jancborchardt** [comment]
@blizzz activity is a differnt discussion for now. This is only about having the sidebar open, with the Sharing tab showing by default.

**Comment by nickvergessen** [comment]
> activity is a differnt discussion for now (can't find the issue). This is only about having the sidebar open, with the Sharing tab showing by default.

https://github.com/nextcloud/activity/issues/45

**Comment by danxuliu** [comment]
>  If the screen width is above a certain value, we could automatically open the sidebar for the folder you are currently in.

What would be the value (or values) to open the sidebar by default? [This is similar to something planned for Nextcloud Talk](https://github.com/nextcloud/spreed/issues/461), so it would be good to behave consistently (if it makes sense, of course).

**Comment by nickvergessen** [comment]
I would show it on full HD already.
> ![bildschirmfoto vom 2017-11-29 um 09 58 43](https://user-images.githubusercontent.com/213943/33366564-e9c9f476-d4eb-11e7-9665-cbc9b1985a52.png)

Content area looks still big enough to me, and an average user doesn't have too long file names, so the 750px of the file name column wouldn't be enough.

**Comment by jancborchardt** [comment]
Yep, on full HD absolutely. @danxuliu can you prepare a PR with that value to start with, then we can test and see what needs to be adjusted?

**Comment by danxuliu** [comment]
@jancborchardt 
>  can you prepare a PR with that value to start with, then we can test and see what needs to be adjusted?

Please see nextcloud/spreed#516 I will work on the PR for server once it is open again for new features... and I have some time ;-)

**Comment by Abijeet** [comment]
I can take a crack at this some at point in the next two weeks,

@nextcloud/designers, Here's what I think we should do if the page size is enough for the details section to be shown,

1. On page load, display the details section with the details of the first file.
2. If I click on a folder, currently the sidebar closes, so after the folder contents are loaded, open the sidebar again displaying the details of the first file.
3. Sorting the files will not change the currently automatically selected file. This means that since we are showing the details of the first file on page load, sorting will continue to show that file details even though it may not be first anymore.
4. If no files are present, details section will remain collapsed.
5. If a folder is empty, and a new file is uploaded, the details section will automatically open for that file.
6. If the file whose details are currently shown is deleted, show the details of the next file.

## Doubts
1. If the user closes the auto-opened details section, the next time he goes into a new folder or refreshes the page should the details section still open?

That's all I can think of right now. Will comment if I have anything else.

**Comment by Abijeet** [comment]
@jancborchardt - Would do you think of my previous comment?

**Comment by MorrisJobke** [comment]
Nothing for 14 it seems -> moved to 15

**Comment by brasofilo** [comment]
Hi folks, the request was removed from Nextcloud 18 milestone and I'm not seeing it on 19 milestones either... 
There's no workaround to have the sidebar permanently opened, is it?

**Comment by skjnldsv** [comment]
Not yet, this is soon coming though.
Will be implemented slowly but surely in all the apps that supports it until all Nextcloud supports it :)

**Comment by schnellschnell** [comment]
I have a project that needs the _comment section_ always open in browsers. Is there a hack or an app or anything to begin with? I'd like to develop something that can be used by anybody later. I have some modest time and funding for that project. Have you tried something or is anybody interested?
